app/models/concerns/client_file_base.rb

```diff
@@ -49,7 +50,7 @@ def tags
     end
 
     def file_exists_and_not_too_large
-      errors.add :client_file, full_message: 'No uploaded file found' if (client_file.byte_size || 0) < 100
+      errors.add :client_file, full_message: 'No uploaded file found. The upload may have expired; please try reloading the page and retrying the upload.' if (client_file.byte_size || 0) < 100
```

Comment:
Sorry about the scope creep. This is in order to give a more graceful ux on form processing in case the signed blob ID is expired. Let me know if you think this change is ok. I played around with just adding to the `file.errors` directly, but it gets cleared out later when `.valid?` is called.
```
f = Hmis::File.last
f.errors.add(:base, :invalid, full_message: 'oh no')
f.errors
=> #<ActiveModel::Errors [#<ActiveModel::Error attribute=base, type=invalid, options={:full_message=>"oh no"}>]>
f.valid?
f.errors
=> #<ActiveModel::Errors []>
```

Reply:
Adding some screenshots from testing for reference. I only tested on the File form so far, not files on assessments.


This is what I see if I try to submit the File form without actually uploading anything. The first error is coming from the Form Definition validation code, because the "Attachment" field is marked as `required: true`. The second error is coming from here. (The issue of duplicated errors here pre-dates this PR).

<img width="920" alt="Screenshot 2024-12-09 at 9 39 44 AM" src="https://github.com/user-attachments/assets/3629384b-2573-4005-b58d-9f6ba92e4317">

This is what I see if I try to upload a blob that has been expired:
(The form required-ness validation isn't present because there was a blob id submitted)

<img width="900" alt="Screenshot 2024-12-09 at 11 23 42 AM" src="https://github.com/user-attachments/assets/902289a9-b558-4ee9-bc81-ee266db6dc75">

Reply:
I don't know if it's worth adding this message since we added such a long expiration window, and most likely the user will see it because they didn't upload a file. But we should address the duplicate error problem separately.

Reply:
Ok, makes sense. I am happy to add a ticket to address that, and revert this change here.

Reply:
Ok lets go with that. thanks

Reply:
Created https://github.com/open-path/Green-River/issues/7075